### PR TITLE
Post: dont rerender all FlatList items

### PR DESCRIPTION
**Pull request by szymonkoper**
A new blogpost

### CODE REVIEW

File: src/pages/markdown/2022-dont-rerender-all-flatlist-items/index.md [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+    return <FormListItem answer={answer} setValue={setValue} />;
+  },
+  []
+);
+```
+
+
+The new `renderItem` that supports normalized data now as a param needs only item id (`answerId`) to render complete `NormalizedFormListItem`. There is also no need for `useCallback` anymore because we can now move it out of the `HomeScreen` component:
+
+```javascript
```
Comment (blazyliszek) [inline src/pages/markdown/2022-dont-rerender-all-flatlist-items/index.md]: I see you're using javascript code block here.
I remember I was using ```tsx and ```ts, but I don't know if that makes any difference. You can check that, if you're bored :)

Comment (blazyliszek) [reply src/pages/markdown/2022-dont-rerender-all-flatlist-items/index.md]: great work btw.

Comment (szymonkoper) [reply src/pages/markdown/2022-dont-rerender-all-flatlist-items/index.md]: Thanks! It makes a difference. Fixed 👍

File: src/pages/markdown/2022-dont-rerender-all-flatlist-items/index.md [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+
+      <FormFooter style={styles.footer} />
+    </SafeAreaView>
+  );
+};
+```
+
+So we did it. 🔥
+
+As soon as the first users received the new app and started using it, the first negative reviews were coming. Some users were complaining that the app was slow when sliding.
```
Comment (sharnik) [inline src/pages/markdown/2022-dont-rerender-all-flatlist-items/index.md]:
```suggestion
As soon as the first users received the new app and started using it, the first negative reviews started coming. Some users were complaining that the app was slow when sliding.
```

File: src/pages/markdown/2022-dont-rerender-all-flatlist-items/index.md [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+Everything green is re-rendering. It's bad.
+
+</div>
+
+
+#### Theory
+
+_React_ re-renders components when _props_ or _state_ change. By default, re-render propagates to all children components. Chart shows the `FlatList` component re-rendered because of the `data` prop change. It's where we pass `answers`.
+
+Here is an example of how the `answers` list (`data` prop) looks like:
```
Comment (sharnik) [inline src/pages/markdown/2022-dont-rerender-all-flatlist-items/index.md]:
```suggestion
Here is an example of how the `answers` array (`data` prop) looks like:
```

File: src/pages/markdown/2022-dont-rerender-all-flatlist-items/index.md [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+
+If we want to change the rate `value` in a single answer, let's say "_Menu easy to read_" from _6_ to _7_, we have to create a copy of the old list with a single item different. It results in a new `answers` object.
+
+In console logs "_Data has changed_" represents `answers` value change, but we see it each time user updates just one of his answers. It's because whenever user slides one list item slider new `answers` list is passed as a new `data` prop value, which results in re-rendering the whole `FlatList` and all it's children. 
+
+However, there is a way to avoid those unnecessary re-renders and save CPU time. It will help slow smartphones to catch up with performance. 🐌
+
+## Let's give data the shape
+
+What if we could change the shape of the `data` list prop passed to `FlatList` to have the same value always?
```
Comment (sharnik) [inline src/pages/markdown/2022-dont-rerender-all-flatlist-items/index.md]:
```suggestion
What if we could change the shape of the `data` prop passed to `FlatList` to have the same value always?
```

File: src/pages/markdown/2022-dont-rerender-all-flatlist-items/index.md [hunk 5]
```diff
@@ -0,0 +1,294 @@
...
+
+In the list `data`, we can store just ids referring to "full" items. It's just a `string[]` list, not a list of complex objects with a `value` field. When the user slides the slider and updates the `value,` it won't change the item `id`. The `data` list will stay unchanged.
```
Comment (sharnik) [inline src/pages/markdown/2022-dont-rerender-all-flatlist-items/index.md]:
```suggestion
Inside `data`, we can store just ids referring to "full" items. That would make it a `string[]` list, not a list of complex objects with a `value` field. When the user slides the slider and updates the `value`, it won't change the item `id`. The `data` list will stay unchanged.
```

**Review by sharnik — APPROVED** [review]

**Pull request merged**